packages/sit-onyx/src/components/OnyxInput/OnyxInput.vue

```diff
@@ -0,0 +1,128 @@
+<script lang="ts" setup>
+import { computed } from "vue";
+import type { OnyxInputProps } from "./types";
+
+const props = withDefaults(defineProps<OnyxInputProps>(), {
+  modelValue: "",
+  type: "text",
+});
+
+const emit = defineEmits<{
+  /**
+   * Emitted when the current value changes.
+   */
+  "update:modelValue": [value: string];
+  /**
+   * Emitted when the current value changes but only when the input is blurred.
+   */
+  change: [value: string];
+  /**
+   * Emitted when the input is focussed.
+   */
+  focus: [];
+  /**
+   * Emitted when the input is blurred.
+   */
+  blur: [];
+}>();
+
+/**
+ * Current value (with getter and setter) that can be used as "v-model" for the native input.
+ */
+const value = computed({
+  get: () => props.modelValue,
+  set: (value) => emit("update:modelValue", value),
+});
+
+const handleChange = (event: Event) => {
+  const inputValue = (event.target as HTMLInputElement).value;
+  emit("change", inputValue);
+};
+</script>
+
+<template>
+  <label class="onyx-input">
+    <p class="onyx-input__label onyx-text--small onyx-truncation-ellipsis">{{ props.label }}</p>
+
+    <div class="onyx-input__wrapper">
+      <input
+        v-model="value"
+        class="onyx-input__native"
+        :placeholder="props.placeholder"
+        :type="props.type"
+        @change="handleChange"
+        @focus="emit('focus')"
+        @blur="emit('blur')"
+      />
+    </div>
+  </label>
+</template>
+
+<style lang="scss">
+.onyx-input {
+  --border-color: var(--onyx-color-base-primary-500);
+  --selection-color: var(--onyx-color-base-primary-200);
+
+  font-family: var(--onyx-font-family);
+
+  &__label {
+    margin-top: 0;
+    margin-bottom: var(--onyx-spacing-5xs);
+    color: var(--onyx-color-text-icons-neutral-medium);
+  }
+
+  $padding-vertical: var(--onyx-spacing-2xs);
+  $line-height: 1.5rem;
+
+  &__wrapper {
+    border-radius: var(--onyx-radius-sm);
+    border: var(--onyx-1px-in-rem) solid var(--onyx-color-base-neutral-300);
+    background: var(--onyx-color-base-background-blank);
+    color: var(--onyx-color-text-icons-neutral-intense);
+
+    display: flex;
+    align-items: center;
+    gap: var(--onyx-spacing-2xs);
+
+    font-size: 1rem;
+    line-height: $line-height;
+
+    height: calc($line-height + 2 * $padding-vertical);
+    box-sizing: border-box;
+
+    &:has(.onyx-input__native:enabled:hover) {
+      border-color: var(--onyx-color-base-primary-400);
+    }
+
+    &:has(.onyx-input__native:enabled:focus) {
+      border-color: var(--border-color);
+      caret-color: var(--border-color);
```

Comment:
I wouldn't change the caret-color, that could create a11y and annoy users.

Reply:
Thats the current design: https://www.figma.com/file/YfEUBOHk4J4nYrk04geswG/onyx---Design-System-Figma-Library?type=design&node-id=1005-170709&mode=dev

Reply:
But would be fine for me to remove the caret color

Reply:
I talked with Jannick, he is fine with removing it